src/lib/db.js

```diff
@@ -38,9 +38,54 @@ function dbinit() {
         });
       });
     });
+  /* Version 5 upgrade summary:
+   * Drop tables activity and creator to allow better naming and integer primary key.
+   * thanks: 'creator' key (which was the url of a creator) replaced by creator_id
+   * donations: 'url' key (which was the url of a creator) replaced by creator_id
+   * creators: 'url' becomes multi-valued
+   */
+  db.version(5)
+    .stores({
+      activity: null,
+      activities: '++id, &url, title, duration, creator_id',
+      creator: null,
+      creators: '++id, *url, name, ignore',
+      thanks: '++id, url, date, title, creator_id',
+      donations: '++id, date, creator_id, weiAmount, usdAmount',
+    })
+    .upgrade(trans => {
+      trans.activity
```

Comment:
Doesn't this need to return a promise or something? Or will the transaction magically know when it and the upgrade is completed?